catalog/plugin/catalog-plugin-metacardingest-network/src/main/java/org/codice/ddf/catalog/plugin/metacard/util/AttributeFactory.java

```
@@ -0,0 +1,193 @@
+ * multi-valued attributes as a single string is to delimit each value with a comma.
+ */
+public class AttributeFactory {
+
+    private static final Logger LOGGER = LoggerFactory.getLogger(AttributeFactory.class);
+
+    private static final String MULTI_VALUED_SPLIT_REGEX = ",";
+
+    /**
+     * Attempts to create an {@link Attribute} according to the provided {@link AttributeDescriptor}
+     * whose value is represented by the given string {@param value}. Throws an exception if
+     * {@param value} could not be parsed.
+     *
+     * @param attributeDescriptor The descriptor used to create the attribute.
+     * @param value               The non-empty string to use to create an attribute value. Multi-valued
+     *                            entities should be separated by commas.
+     * @return An {@link Attribute} conforming to the given {@link AttributeDescriptor} with a value
+     * of {@param value} that was parsed and processed to {@link Serializable}.
+     */
+    public Attribute createAttribute(AttributeDescriptor attributeDescriptor, String value) {
+        notNull(attributeDescriptor);
+        notEmpty(value);
+
+        if (attributeDescriptor.isMultiValued()) {
+            LOGGER.trace("Found multi-valued attribute descriptor {} for value {}",
+                    attributeDescriptor.getName(),
+                    value);
+            List<String> values = handleMultiValuedString(value);
+            List<Serializable> serializables = values.stream()
+                    .map(string -> parseAttributeValue(attributeDescriptor, string))
+                    .filter(Objects::nonNull)
+                    .map(Serializable.class::cast)
+                    .collect(Collectors.toList());
+            notEmpty(serializables);
+            return doCreate(attributeDescriptor, serializables);
+        }
+
+        Serializable attributeValue = parseAttributeValue(attributeDescriptor, value);
+        notNull(attributeValue);
```

The approach I took was "at least one value must succeed". In the event of a single-value attribute, if that `Serializable` is not valid, we don't have much of a choice and should fail. With multi-valued attributes we behave greedily and only fail if we can't parse **any** of the values. 

Since this might confuse admins if they expect one result but get another due to a parsing error, let's change it to fail fast.